src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -417,22 +417,15 @@ public bool Blocking
             }
         }
 
+        // On .NET Framework, this property functions as a socket-level switch between IOCP-based and Win32 event based async IO.
+        // On that platform, setting UseOnlyOverlappedIO = true prevents assigning a completion port to the socket,
+        // allowing calls to DuplicateAndClose() even after performing asynchronous IO.
+        // .NET (Core) Windows sockets are entirely IOCP-based, and the concept of "overlapped IO" 
+        // does not exist on other platforms, therefore UseOnlyOverlappedIO is a dummy, compat-only property.
         public bool UseOnlyOverlappedIO
         {
-            get
-            {
-                return false;
-            }
-            set
-            {
-                //
-                // This implementation does not support non-IOCP-based async I/O on Windows, and this concept is
-                // not even meaningful on other platforms.  This option is really only functionally meaningful
-                // if the user calls DuplicateAndClose.  Since we also don't support DuplicateAndClose,
-                // we can safely ignore the caller's choice here, rather than breaking compat further with something
-                // like PlatformNotSupportedException.
-                //
-            }
+            get { return false; }
```

Comment:
Do we have any tests that validate it defaults to false and that the setter is a nop?

Reply:
Yes:
https://github.com/dotnet/runtime/blob/1edc35f8517bbb2fae8c0152de14bd105f9bac06/src/libraries/System.Net.Sockets/tests/FunctionalTests/SocketDuplicationTests.cs#L30-L38